.github/workflows/mfem-sanitizer.yml

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+# at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+# LICENSE and NOTICE for details. LLNL-CODE-806117.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability visit https://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the BSD-3 license. We welcome feedback and contributions, see file
+# CONTRIBUTING.md for details.
+
+name: "Sanitizer"
+
+permissions:
+  actions: write
+
+on:
+  push:
+    branches:
+      - master
+      - next
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  Serial:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.11.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: MFEM Checkout
+      uses: actions/checkout@v3
+      with:
+        path: mfem
+
+    - name: MFEM Build
+      uses: mfem/github-actions/build-mfem@v2.3
+      with:
+        os: ${{ runner.os }}
+        target: opt
+        mpi: seq
+        hypre-dir: unused-hypre-dir
+        metis-dir: unused-metis-dir
+        mfem-dir: mfem
+        build-system: make
+        library-only: false
+        config-options: CXX=clang++-14 CXXFLAGS="-O1 -std=c++11 -fsanitize=address -fsanitize-address-use-after-scope -fno-omit-frame-pointer"
+
+    - name: MFEM Info
+      working-directory: mfem
+      run: make info
+
+    - name: MFEM Sanitize
+      working-directory: mfem
+      run: ASAN_OPTIONS=strict_string_checks=1:detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1,detect_leaks=1,halt_on_error=1 make test
```

Comment:
Is there a way to wrap the long lines? (And similarly for line 51)

Reply:
I think this should work: https://github.com/pazner/mfem/blob/cb1edb56d882e2c062b24f46db9eaa723c723e17/.github/workflows/mfem-sanitizer.yml

Reply:
Actually, the comment after `config-options:` seems redundant, so this should work fine too:
```yaml
        config-options:
          CXX="clang++-14"
          CXXFLAGS="-O1 -std=c++11
            -fsanitize=address
            -fsanitize-address-use-after-scope
            -fno-omit-frame-pointer"
```

Reply:
The `run` seems to work like this:
```yaml
      run:
        ASAN_OPTIONS=\
          strict_string_checks=1:\
          detect_stack_use_after_return=1:\
          check_initialization_order=1:\
          strict_init_order=1,\
          detect_leaks=1,\
          halt_on_error=1
        make test
```

Reply:
Is `ASAN_OPTIONS` sensitive to the added spaces (from replacing the newlines)?

The version I posted [earlier](https://github.com/pazner/mfem/blob/cb1edb56d882e2c062b24f46db9eaa723c723e17/.github/workflows/mfem-sanitizer.yml) looks like it works ([ran successfully here](https://github.com/pazner/mfem/actions/runs/4663643278/jobs/8255148793)).

Reply:
> The `run` seems to work like this:
> 
> ```yaml
>       run:
>         ASAN_OPTIONS=\
>           strict_string_checks=1:\
>           detect_stack_use_after_return=1:\
>           check_initialization_order=1:\
>           strict_init_order=1,\
>           detect_leaks=1,\
>           halt_on_error=1
>         make test
> ```

Don't you want `make test` on the same line as the `ASAN_OPTIONS` assignment, otherwise `make` won't see the environment variable?

Or I suppose you could add `export ASAN_OPTIONS=…`.

Reply:
The resulting string it produced was: `ASAN_OPTIONS=\ strict_string_checks=1:\ detect_stack_use_after_return=1:\ check_initialization_order=1:\ strict_init_order=1,\ detect_leaks=1,\ halt_on_error=1 make test`, but you're right, it might be cleaner to use the block.